module/optee/smt/test/mod_optee_smt_unit_test.c

```diff
@@ -0,0 +1,889 @@
+    status = smt_get_message_header(channel_id, &header);
+    TEST_ASSERT_EQUAL(FWK_E_ACCESS, status);
+}
+
+/*!
+ * \brief test get_message_header function.
+ *
+ *  \details Test all other cases for get_message_header function.
+ */
+void test_smt_get_message_header(void)
+{
+    int status;
+    fwk_id_t channel_id;
+    uint32_t header;
+
+    /* Return header */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    header = 0x0000;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_get_message_header(channel_id, &header);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(((struct mod_optee_smt_memory *)(mailbox_virt_buffer))->message_header, header);
+}
+
+/*!
+ * \brief test early access to get_payload function.
+ *
+ *  \details Test all cases for get_payload function before
+ *  a message has been received.
+ */
+void test_smt_get_payload_early(void)
+{
+    int status;
+    fwk_id_t channel_id;
+    void *fake_payload;
+
+    /* Fail if the pointer is null */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_get_payload(channel_id, NULL, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_PARAM, status);
+
+    /* No message available */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_get_payload(channel_id, (const void **)&fake_payload, NULL);
+    TEST_ASSERT_EQUAL(FWK_E_ACCESS, status);
+}
+
+/*!
+ * \brief test get_payload function.
+ *
+ *  \details Test all remaining cases for get_payload function.
+ */
+void test_smt_get_payload(void)
+{
+    int status;
+    fwk_id_t channel_id;
+    void *fake_payload;
+    size_t fake_size;
+
+    /* Return payload pointer */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_get_payload(channel_id, (const void **)&fake_payload, NULL);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(((struct mod_optee_smt_memory *)(channel_in_buffer))->payload, fake_payload);
+
+    /* Return payload pointer and size */
+    channel_id = FWK_ID_ELEMENT(FWK_MODULE_IDX_OPTEE_SMT, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    fake_size = 0x0000;
+    fwk_id_get_element_idx_ExpectAndReturn(channel_id, FAKE_SCMI_SERVICE_IDX_OSPM_0);
+    status = smt_get_payload(channel_id, (const void **)&fake_payload, &fake_size);
+    TEST_ASSERT_EQUAL(FWK_SUCCESS, status);
+    TEST_ASSERT_EQUAL(((struct mod_optee_smt_memory *)(channel_in_buffer))->payload, fake_payload);
+    TEST_ASSERT_EQUAL(((struct mod_optee_smt_memory *)(channel_in_buffer))->length - sizeof(uint32_t), fake_size);
+
+}
+
+void fake_fwk_str_memcpy(void *dest, const void *src, size_t count, int iteration)
```

Comment:
Minor:
As I understand from your PR: you try to make helper/callback functions before usage
However, I don't see it in `test_smt_write_payload_early` So could you move it before `test_smt_write_payload`?